opt-in.md

```diff
@@ -1,96 +1,115 @@
+* Have no stored credentials (for prefetch) or storage of any kind (for prerender); or
+* Indicate that they are prepared to perform this sort of upgrade, by sending the appropriate `Supports-Loading-Mode` header value: either `uncredentialed-prefetch`, `uncredentialed-prerender`, or both.
 
-<dl>
-  <dt><code>default</code></dt>
-  <dd>Implied, even if not listed. Baseline standard behavior.</dd>
+_Note: these values are not currently implemented in Chromium._
 
-  <dt><code>uncredentialed-prefetch</code></dt>
-  <dd>The resource is suitable for any user who meets the cache conditions. Either it is not personalized, or it includes script to modify the document to reflect credentialed state (e.g., to show the user's logged-in state). However, access to credentials will be available when the document loads and script executes.</dd>
+### Cross-origin same-site prerendering
 
-  <dt><code>uncredentialed-prerender</code></dt>
-  <dd>Implies <code>uncredentialed-prefetch</code>. The resource can be loaded without access to its credentials and storage. Access to certain other APIs may be limited in this state. Either it is not personalized based on credentials, or it includes script to modify the document when the loading state changes to permit it.</dd>
-</dl>
+For prerendering, there is an additional complication when dealing with cross-origin destinations. Because the web's privacy boundary is site, we can send credentials and give storage access to such pages before activation, with no privacy issues. However, the web's _security_ boundary is the origin, not the site. So we need to ensure that pages cannot use prerendering to attack other origins—even same-site ones.
 
-The `<meta>` tag is processed only if it appears within the `<head>` element and no `<script>`, `<noscript>` or `<template>` tag appears before it.
-This means that the supported loading modes, if not declared in a response header, can be statically computed with use of an HTML parser without rendering or script execution.
-See [the meta processing model](meta-processing.md) for details.
+As a concrete example of what an "attack" might look like, consider an organization which provides both web hosting and document editing services. Furthermore, assume that whenever you access a document on the document editing service, it updates the user's list of recently-viewed documents. And finally, assume that the document editing service is not prerendering-aware, i.e. it has no code that checks [`document.prerendering`](./prerendering-state.md) to customize its behavior. Then, if we allowed `https://sites.example.com/bobs-star-wars-fan-site` to prerender `https://docs.example.com/public-list-of-best-star-wars-movies` with no opt-in, then Bob's Star Wars Fan Site would be able to update the user's list of recently-viewed documents, pushing the List of Best Star Wars Movies document to the top.
```

Comment:
Hmmmmm, this depends on whether this change is made as part of the script execution or as part of serving the response. In the latter case using something like an `<iframe>` or a non-CORS `fetch` would do the trick, as would a prefetch, unless the server was checking the request headers.

Not a problem per se, but kinda muddies the waters about why this same argument doesn't apply there.

Reply:
Yeah, I was unsure how much detail to go into here. The other document mentions

> Side effects from simply issuing credentialed requests to the target URL, e.g. triggering non-idempotent GETs, are not a concern. These are already possible today via `<iframe>`s, `<img>`s, etc.

which is relevant to what you're saying.

Maybe I'll tweak the wording a bit to try to make it clearer I am envisioning a client-side update...